contracts/modules/Experimental/Checkpoints/WeightedVoteCheckpoint.sol

```diff
@@ -0,0 +1,222 @@
+pragma solidity ^0.5.0;
+
+import "../../Module.sol";
+import "../../Checkpoint/ICheckpoint.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title Checkpoint module for token weighted vote
+ * @notice This voting system uses public votes
+ * @notice In this module every token holder has voting right (Should be greater than zero)
+ * Tally will be calculated as per the weight (balance of the token holder)
+ */
+contract WeightedVoteCheckpoint is ICheckpoint, Module {
+    using SafeMath for uint256;
+
+    struct Ballot {
+        uint256 checkpointId;
+        uint256 totalSupply;
+        uint64 startTime;
+        uint64 endTime;
+        uint64 totalNumVotes;
+        uint56 totalProposals;
+        uint8 isActive;
+        mapping(uint256 => uint256) proposalToVote;
+        mapping(address => uint256) voteByAddress;
+    }
+
```

Comment:
This is really picky (sorry), but I think these variable names could be cleaned up a bit to make things clearer. I'd suggest:
```
    struct Ballot {
        uint256 checkpointId;
        uint256 totalSupply;
        uint64 startTime;
        uint64 endTime;
        uint64 totalVoters;
        uint56 totalProposals;
        uint8 isActive;
        mapping(uint256 => uint256) proposalCount; //or proposalToVotes
        mapping(address => uint256) investorProposal; //or investorToProposal
    }
```